dss/__init__.py

```diff
@@ -57,6 +59,24 @@ def common_error_handler(exception):
         ))
 
 
+@draft4_format_checker.checks('RFC3339')
+def is_RFC3339(val):
+    from iso8601 import iso8601
+    # convert it to date-time so we can format exactly as the system requires (with microsecond precision)
+    version_error = DSSException(
+        requests.codes.bad_request,
+        "illegal_version",
+        f"version should be an rfc3339-compliant timestamp")
+    try:
+        timestamp = iso8601.parse_date(val)
+    except iso8601.ParseError:
+        raise version_error
+    timestamp = datetime_to_version_format(timestamp)
+    if timestamp != val:
+        raise version_error
```

Comment:
So since we're going down this rabbit hole, the error string on L69 is no longer accurate.  L73 would imply a non-RFC3339-compliant timestamp, and L76 would imply a non-HCA-compliant timestamp.

I highly encourage you to consider tweaking the message.  It can just be a "non-HCA-compliant", which accurately describes both errors, but the current error message is not accurate.